### PR TITLE
fix(python): Fix Reference Tracking for `Class` and `Method` AST Nodes

### PR DESCRIPTION
**Pull request by noanflaherty**
## Description
Python Classes and Methods weren't correctly tracking their internal references. This PR fixes these issues.

## Changes Made
- Update constructor of `Class` AST Node
- Update constructor of `Method` AST Node

## Testing
- [ X] Unit tests added/updated

### CODE REVIEW

File: generators/python-v2/ast/src/Class.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export class Class extends AstNode {
         this.extends_.forEach((parentClassReference) => {
             this.inheritReferences(parentClassReference);
         });
+
+        this.decorators.forEach((decorator) => {
+            this.inheritReferences(decorator);
```
Comment (dvargas92495) [inline generators/python-v2/ast/src/Class.ts:40]: [q] the decorator could be a function with arguments. Does this catch those parameters?

File: generators/python-v2/ast/src/Method.ts [hunk 2]
```diff
@@ -50,6 +50,20 @@ export class Method extends AstNode {
         this.type = type;
         this.decorators = decorators ?? [];
         this.static_ = static_ ?? false;
+
+        this.parameters.forEach((parameter) => {
+            this.inheritReferences(parameter);
```
Comment (dvargas92495) [inline generators/python-v2/ast/src/Method.ts:55]: [q] do parameters automatically import their type annotations if necessary?

**Review by dvargas92495 — COMMENTED** [review]
👍

**Review by amckinney — APPROVED** [review]
Assuming @dvargas92495's comments are addressed, this lgtm

**Pull request merged**